### PR TITLE
ALL : Keep adaptive links only

### PR DESCRIPTION
**Pull request by BellezaEmporium**
We talked about this for some time, to prevent using chunklists, we prefer keeping adaptive links (playlist.m3u8, index.m3u8, master.m3u8 etc etc...) that can permit to get the best quality for your bandwidth (depending on what your bandwidth can take, you'll get the equivalent quality to your bandwidth.)

This PR will try to change all the links to keep only playlists.

PS : If I find that the playlist only contains one single line to a chunklist, then I keep the chunklist instead of a playlist that redirects to the same chunklist.

### CODE REVIEW

**Comment by freearhey** [comment]
I suggest converting PR to "Draft" to let everyone know that it is not yet ready.

File: channels/ae.m3u [hunk 1]
```diff
@@ -233,8 +159,6 @@ https://weyyak-live.akamaized.net/weyyak_drama/index.m3u8
 https://weyyak-live.akamaized.net/weyyak_mix/index.m3u8
 #EXTINF:-1 tvg-id="WeyyakNawaem.ae" tvg-name="Weyyak Nawaem" tvg-country="AE" tvg-language="Arabic" tvg-logo="https://content.weyyak.com/d867c2e9-328d-ea11-831d-02d662556796/poster-image" group-title="",Weyyak Nawaem (720p)
 https://weyyak-live.akamaized.net/weyyak_nawaem/index.m3u8
-#EXTINF:-1 tvg-id="Yas.ae" tvg-name="Yas" tvg-country="AE" tvg-language="" tvg-logo="" group-title="",Yas (1080p)
-https://admdn1.cdn.mangomolo.com/yastv/smil:yastv.stream.smil/playlist.m3u8?adtv
```
Comment (Dum4G) [inline channels/ae.m3u:237]: Why, it works

**Review by Dum4G — CHANGES_REQUESTED** [review]
Why? that works. Overall, I see some channels being removed without a replacement. That's not our way to manage things

File: channels/ae.m3u (hunk 1 above)
Comment (LaneSh4d0w) [reply channels/ae.m3u:237]: Duplicate of another one.

Comment (LaneSh4d0w) [reply channels/ae.m3u:237]: > 
> 
> Why? that works. Overall, I see some channels being removed without a replacement. That's not our way to manage things

They were basically the same channels, but only a different chunklist. All of them work, but we keep only one adaptive link that simply do the same thing as keeping 5 or 6 chunk links.

Comment (Dum4G) [reply channels/ae.m3u:237]: I see only one Yas here and it shows Yas icon on the channel

Comment (LaneSh4d0w) [reply channels/ae.m3u:237]: Line 163.

Comment (Dum4G) [reply channels/ae.m3u:237]: Also, we should retain all logos, groups etc if at least a single entry had them

Comment (LaneSh4d0w) [reply channels/ae.m3u:237]: That's what I try to do.

**Comment by Dum4G** [comment]
You can safely keep links that contain just "chunklist.m3u", sometimes it is the same thing as "playlist.m3u"
Ekta is a good example if you shorten the link

**Comment by LaneSh4d0w** [comment]
> 
> 
> You can safely keep links that contain just "chunklist.m3u", sometimes it is the same thing as "playlist.m3u"
> Ekta is a good example if you shorten the link

Ekta was a wrongly named Dubai One, if I remember well.

**Comment by Dum4G** [comment]
RTRS should go live in the next hour

**Comment by LaneSh4d0w** [comment]
> RTRS should go live in the next hour

Thanks for telling me. I'll check if the links in playlist are litterally the ones that exist.

**Comment by Dum4G** [comment]
I can confirm the website is offering me this link but I have CORS error while accessing it. I suppose it's geoblocked
https://rtrs.tv/program/uzivo.php?c=1&bcast_on=1

**Comment by LaneSh4d0w** [comment]
> 
> 
> I can confirm the website is offering me this link but I have CORS error while accessing it. I suppose it's geoblocked
> https://rtrs.tv/program/uzivo.php?c=1&bcast_on=1

Same for me, i'll try using a proxy to see if it works.

**Comment by LaneSh4d0w** [comment]
I agree, it's geo-blocked to Bulgaria.

**Comment by Dum4G** [comment]
I don't think this job should be done by hand. Let me play around with REGEXP a bit and gather as much cases of non-adaptive links as possible. If everything will be performed right we'll have hundred of duplicate links that will be removed the same day automatically. You can merge this the way it is I guess

**Comment by LaneSh4d0w** [comment]
> 
> 
> I don't think this job should be done by hand. Let me play around with REGEXP a bit and gather as much cases of non-adaptive links as possible. If everything will be performed right we'll have hundred of duplicate links that will be removed the same day automatically. You can merge this the way it is I guess

I suppose... You should also verify that links are working at the same instance. There might be instances in which "master.m3u8" only works, instances where only "index.m3u8" works, or even instances where only "playlist.m3u8" works.

**Comment by Dum4G** [comment]
That's why we don't auto-merge things. I will try to handle this based on a source where the link comes from

**Comment by LaneSh4d0w** [comment]
This is a first set of many more PRs to come.

File: channels/bo.m3u [hunk 2]
```diff
@@ -9,29 +9,13 @@ https://mediacp.hostradios.com.ar:19360/atbcochabamba/atbcochabamba.m3u8
 http://boliviatv1.srfms.com:5735/live/livestream/playlist.m3u8
 #EXTINF:-1 tvg-id="BoliviaTV72.bo" tvg-name="Bolivia TV 7.2" tvg-country="BO" tvg-language="Spanish" tvg-logo="https://i.imgur.com/1zTLGdj.jpg" group-title="",Bolivia TV 7.2 (720p)
 https://video1.getstreamhosting.com:1936/8224/8224/chunklist_w1684483949.m3u8
-#EXTINF:-1 tvg-id="BoliviaTV72.bo" tvg-name="Bolivia TV 7.2" tvg-country="BO" tvg-language="Spanish" tvg-logo="https://i.imgur.com/1zTLGdj.jpg" group-title="",Bolivia TV 7.2 (720p)
-https://video1.getstreamhosting.com:1936/8224/8224/playlist.m3u8
```
Comment (Dum4G) [inline channels/bo.m3u:13]: works

File: channels/br.m3u [hunk 3]
```diff
@@ -2,18 +2,12 @@
 #EXTINF:-1 tvg-id="AgroBrasilTV.br" tvg-name="AgroBrasil TV" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://www.costadosol.tv.br/site/wp-content/uploads/2017/12/agro-brasil.png" group-title="Outdoor",AgroBrasil TV (486p)
 http://45.162.230.234:1935/agrobrasiltv/agrobrasiltv/playlist.m3u8
 #EXTINF:-1 tvg-id="AllSports.br" tvg-name="All Sports" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/wULpnYR.png" group-title="Sports",All Sports
-https://5cf4a2c2512a2.streamlock.net/dgrau/dgrau/chunklist.m3u8
+https://5cf4a2c2512a2.streamlock.net/dgrau/dgrau/playlist.m3u8
 #EXTINF:-1 tvg-id="Animestation.br" tvg-name="Animestation" tvg-country="BR" tvg-language="" tvg-logo="https://i.imgur.com/5UpjGcL.png" group-title="",Animestation (480p) [Not 24/7]
 https://stmv.video.expressolider.com.br/animestation1/animestation1/chunklist_w2074350440.m3u8
-#EXTINF:-1 tvg-id="Animestation.br" tvg-name="Animestation" tvg-country="BR" tvg-language="" tvg-logo="https://i.imgur.com/5UpjGcL.png" group-title="",Animestation (480p) [Not 24/7]
-https://stmv.video.expressolider.com.br/animestation1/animestation1/playlist.m3u8
```
Comment (Dum4G) [inline channels/br.m3u:9]: works

File: channels/br.m3u [hunk 4]
```diff
@@ -101,28 +87,24 @@ https://v4-slbps-sambavideos.akamaized.net/live/3282%2C8114%2Cec4b5a296d97fa99bf
...
+https://dhxt2zok2aqcr.cloudfront.net/live/rs21_360p614k/index.m3u8
 #EXTINF:-1 tvg-id="RedeSeculo21.br" tvg-name="Rede Século 21" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/vvrCWVo.jpg" group-title="Religious",Rede Século 21 (240p) [Not 24/7]
 https://d1gs793b1l6k9i.cloudfront.net/out/v1/a3aac39375df40a1a564635d7b41e90c/index.m3u8
 #EXTINF:-1 tvg-id="RedeTVES.br" tvg-name="Rede TV! ES" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://www.livetvnews.com.br/wp-content/uploads/2019/03/Logo-RedeTV.png" group-title="General",Rede TV! ES
 https://hls.brasilstream.com.br/live/redetves/redetves/tracks-v1a1/mono.m3u8
-#EXTINF:-1 tvg-id="RedeVida.br" tvg-name="Rede Vida" tvg-country="BR" tvg-language="" tvg-logo="https://i.imgur.com/rJ0kO2o.png" group-title="",Rede Vida (720p)
-https://cvd1.cds.ebtcvd.net/live-redevida/smil:redevida.smil/chunklist.m3u8
 #EXTINF:-1 tvg-id="Rede Vida" tvg-name="Rede Vida" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/h4WaIw5.jpg" group-title="Religious",Rede Vida (720p)
```
Comment (Dum4G) [inline channels/br.m3u:98]: bad tvg id

File: channels/br.m3u [hunk 5]
```diff
@@ -101,28 +87,24 @@ https://v4-slbps-sambavideos.akamaized.net/live/3282%2C8114%2Cec4b5a296d97fa99bf
...
-https://dhxt2zok2aqcr.cloudfront.net/live/rs21_360p614k/index.m3u8
 #EXTINF:-1 tvg-id="RedeSuper.br" tvg-name="Rede Super" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/3ODbxcM.png" group-title="Religious",Rede Super (432p)
 https://api.new.livestream.com/accounts/10205943/events/3429501/live.m3u8
+#EXTINF:-1 tvg-id="RedeSeculo21.br" tvg-name="Rede Século 21" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/vvrCWVo.jpg" group-title="Religious",Rede Século 21
+https://dhxt2zok2aqcr.cloudfront.net/live/rs21_360p614k/index.m3u8
 #EXTINF:-1 tvg-id="RedeSeculo21.br" tvg-name="Rede Século 21" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/vvrCWVo.jpg" group-title="Religious",Rede Século 21 (240p) [Not 24/7]
 https://d1gs793b1l6k9i.cloudfront.net/out/v1/a3aac39375df40a1a564635d7b41e90c/index.m3u8
 #EXTINF:-1 tvg-id="RedeTVES.br" tvg-name="Rede TV! ES" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://www.livetvnews.com.br/wp-content/uploads/2019/03/Logo-RedeTV.png" group-title="General",Rede TV! ES
 https://hls.brasilstream.com.br/live/redetves/redetves/tracks-v1a1/mono.m3u8
```
Comment (Dum4G) [inline channels/br.m3u:97]: redundant tracks-v1a1

File: channels/br.m3u [hunk 6]
```diff
@@ -286,19 +264,15 @@ http://flash.softhost.com.br:1935/ufg/tvufgweb/playlist.m3u8
...
 https://14398c.ha.azioncdn.net/primary/smil:tv_universal.smil/playlist.m3u8
 #EXTINF:-1 tvg-id="TVVerdesCampos.br" tvg-name="TV Verdes Campos" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/1mK39TC.png" group-title="General",TV Verdes Campos
-https://596639ebdd89b.streamlock.net/8124/8124/chunklist.m3u8
+https://596639ebdd89b.streamlock.net/8124/8124/playlist.m3u8
 #EXTINF:-1 tvg-id="TVCI.br" tvg-name="TVCI" tvg-country="BR" tvg-language="" tvg-logo="" group-title="",TVCI (360p)
 http://flash8.crossdigital.com.br:1935/2306/2306/live.m3u8
 #EXTINF:-1 tvg-id="TVCIParanagua.br" tvg-name="TVCI Paranaguá" tvg-country="BR" tvg-language="Portuguese" tvg-logo="https://i.imgur.com/4GKZbXD.png" group-title="General",TVCI Paranaguá (360p)
 http://flash8.crossdigital.com.br/2306/2306/playlist.m3u8
-#EXTINF:-1 tvg-id="TVE.br" tvg-name="TVE" tvg-country="BR" tvg-language="" tvg-logo="" group-title="",TVE
-http://stream2.ba.gov.br:80/hls-live/livepkgr/_definst_/irdeb/pgm-1.m3u8
```
Comment (Dum4G) [inline channels/br.m3u:301]: works

**Review by Dum4G — CHANGES_REQUESTED** [review]

File: channels/bo.m3u (hunk 2 above)
Comment (LaneSh4d0w) [reply channels/bo.m3u:13]: It's because the playlist only has got the chunklist above, not useful to keep it.

File: channels/br.m3u (hunk 3 above)
Comment (LaneSh4d0w) [reply channels/br.m3u:9]: It's because the playlist only has got the chunklist above, not useful to keep it.

File: channels/br.m3u (hunk 4 above)
Comment (LaneSh4d0w) [reply channels/br.m3u:98]: Will need to fix it

File: channels/br.m3u (hunk 5 above)
Comment (LaneSh4d0w) [reply channels/br.m3u:97]: It's because the playlist file only has got this version, not useful to keep it.

File: channels/br.m3u (hunk 6 above)
Comment (LaneSh4d0w) [reply channels/br.m3u:301]: I think I had changed this version. Don't quite remember for this one.

Comment (Dum4G) [reply channels/br.m3u:301]: Any ideas?

Comment (LaneSh4d0w) [reply channels/br.m3u:301]: On retest it works. Maybe I was trying to search for a better version but went out.

Reinstating it.

**Review by Dum4G — APPROVED** [review]

**Pull request merged**